actioncable/lib/action_cable/channel/base.rb

```diff
@@ -189,11 +189,15 @@ def perform_action(data)
       # This method is called after subscription has been added to the connection and
       # confirms or rejects the subscription.
       def subscribe_to_channel
-        run_callbacks :subscribe do
-          subscribed
-        end
+        run_callbacks :subscribe, :before
+        return reject_subscription if subscription_rejected?
+
+        subscribed
+        return reject_subscription if subscription_rejected?
```

Comment:
Currently, Action Cable executes the `after_subscribe` callbacks after `#subscribed` [even if the subscription was rejected](https://github.com/rails/rails/blob/v7.2.0/actioncable/lib/action_cable/channel/callbacks.rb#L52-L53). Doesn't this line break the current behaviour?

Reply:
Hmm nice catch, I guess we should only return if `#subscribed` hasn't been called yet.

Reply:
I've updated the logic and added some coverage for this case.